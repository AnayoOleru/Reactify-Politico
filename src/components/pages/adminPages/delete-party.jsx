import React, { Component } from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import store from '../../../store';
import AdminNavBar from '../../reuseable component/admin-navbar.component';
import swal from 'sweetalert';
import { deleteParty } from '../../../actions/deleteAction';
import '../../../styles/addParties-style.css';
import '../../../styles/admin-modal.style.css';


export class DeleteParty extends Component {
  constructor(props) {
    super(props);
    this.state = {
      partyname: '',
    };
    this.onChange = this.onChange.bind(this);
  }


  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  deletePartyHandler = async (e) => {
    e.preventDefault();

    const partyId = this.props.match.params.id;

    await this.props.deleteParty(partyId);
    swal({
      icon: 'success',
      title: `Party deleted successfully`,
    });
    // window.location = '/all-parties';
  }

  render() {

    const style4 = {
      backgroundColor: '#ffffff',
    };

    const {
      partyname,
    } = this.state;

    return (
      <Provider store={store}>
        <React.Fragment>
          <AdminNavBar />
          <main style={style4}>
            <section className="section-cards">
              <div className="text-cards">
                <h1 className="heading-primary">
                  ARE YOU SURE YOU SURE?
                        </h1>
              </div>

              <form className="entry-form">
                <footer className="entry-footer">
                  <input className="entry-button" type="submit" name="btn_signin" value="Yes" onClick={this.deletePartyHandler} />
                  <br />
                  <br />
                  <input className="entry-button" type="submit" name="btn_signin" value="Cancel" href={'/add-parties'} />
                </footer>
              </form>
            </section>
          </main>
        </React.Fragment>
      </Provider>
    );
  }
}

DeleteParty.propTypes = {
  deleteParty: PropTypes.func.isRequired,
}

const mapStateToProps = (state, posts) => ({
  party: state.get.items.data,
  error: posts.error,
});

export default connect(mapStateToProps, { deleteParty })(DeleteParty);
