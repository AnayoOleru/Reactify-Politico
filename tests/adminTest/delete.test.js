import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { DeleteParty } from '../../src/components/pages/adminPages/delete-party.jsx';
import AdminNavbar from '../../src/components/reuseable component/admin-navbar.component.jsx';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({});

const OfficeComponent = (
  <Provider store={store}>
    <DeleteParty />
  </Provider>
);

describe('Alloffice component', () => {
     it('render component', () => {
         const wrapper = shallow(<DeleteParty />);
     });
});

describe('Office component', () => {
  it('should match snapshot', () => {
    const wrap = OfficeComponent;
    expect(wrap).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const wrapper = shallow(OfficeComponent);
    expect(wrapper.find(<DeleteParty />));
  });

  it('should render without crashing', () => {
    const wrapper = shallow(OfficeComponent);
    expect(wrapper.find(<AdminNavbar />));
  });
});
