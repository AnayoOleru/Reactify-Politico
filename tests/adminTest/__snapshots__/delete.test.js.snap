// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Office component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UNDEFINED />
</Provider>
`;
